library/ssl_srv.c

```diff
@@ -1322,8 +1322,8 @@ static int ssl_parse_client_hello_v2( mbedtls_ssl_context *ssl )
     for( i = 0, p = buf + 6; i < ciph_len; i += 3, p += 3 )
     {
         if( p[0] == 0 &&
-            p[1] == (unsigned char)( ( MBEDTLS_SSL_FALLBACK_SCSV_VALUE >> 8 ) & 0xff ) &&
-            p[2] == (unsigned char)( ( MBEDTLS_SSL_FALLBACK_SCSV_VALUE      ) & 0xff ) )
+            p[1] == MBEDTLS_BYTE_1( MBEDTLS_SSL_FALLBACK_SCSV_VALUE ) &&
+            p[2] == MBEDTLS_BYTE_0( MBEDTLS_SSL_FALLBACK_SCSV_VALUE ) )
```

Comment:
Could replace with `MBEDTLS_GET_UINT16_BE(p, 1) == MBEDTLS_SSL_FALLBACK_SCSV_VALUE` for more readability?

Reply:
I might not understand this part of the code correctly, but say for the sake of example we are checking `MBEDTLS_SSL_FALLBACK_SCSV_VALUE == 0x01` and p[1] and p[2] are both 0x01, would that not make `MBEDTLS_GET_UINT16_BE(p, 1)` 0x0101? rather than comparing 0x01 to 0x01 we would be comparing 0x01 to 0x0101?

Reply:
I think not quite, because `MBEDTLS_BYTE_1(0x01) == MBEDTLS_BYTE_1(0x0001) == 0x00` so in the current code we'd be checking that `p[1] == 0x00` and `p[2] == 0x01`. Does that make vague sense?

Reply:
oh yes I see, not sure how I misunderstood that, will get on it